includes/Core/Modules/REST_Modules_Controller.php

```diff
@@ -755,8 +755,8 @@ private function prepare_module_data_for_response( Module $module ) {
 			'internal'     => $module->internal,
 			'order'        => $module->order,
 			'forceActive'  => $module->force_active,
-			'shareable'    => $module->is_shareable(),
```

Comment:
While the changes suggested in the IB are perfectly understandable, that unfortunately doesn't solve the entire problem because the module is being declared shareable here.

Since the `PageSpeed_Insights` module doesn't overwrite the `is_connected` method, it inherits it from the `Module` class which is `true` by default. As a result, `$module->is_shareable()` also results `true` for the PSI module.

Reply:
I agree. A potential solution could be introducing a `Module::is_active` class, which would receive its value from the DB option. We could then use that to return `false` for `Module::is_connected` if it was not active.

Reply:
Opened #7274 to discuss further about this.